### PR TITLE
Add support for accounts with more than 8 chars

### PR DESCRIPTION
**Pull request by imbalanced**
OOG - Updated findCharacter, getCharacters, and loginCharacter functions to handle up to 24 characters.

### CODE REVIEW

File: d2bs/kolbot/libs/OOG.js [hunk 1]
```diff
@@ -900,18 +901,49 @@ MainLoop:
...
-					if (text instanceof Array && typeof text[1] === "string" && text[1] === info.charName) {
-						return true;
+					if (text instanceof Array && typeof text[1] === "string") {
+						count++;
+
+						if (text[1] === info.charName) {
+							return true;
+						}
+
+						if (count >= 24) {
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js]: consider moving to while loop:
`} while (count < 24 && control.getNext());`

File: d2bs/kolbot/libs/OOG.js [hunk 2]
```diff
@@ -900,18 +901,49 @@ MainLoop:
...
+
+					control.click();
+					sendKey(0x28);
+					sendKey(0x28);
+					sendKey(0x28);
+					sendKey(0x28);
+
+					me.blockMouse = false;
+
+					continue;
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js]: Is the continue and delay necessary?

File: d2bs/kolbot/libs/OOG.js [hunk 3]
```diff
@@ -975,15 +1041,33 @@ MainLoop:
...
 						text = control.getText();
 
-						if (text instanceof Array && typeof text[1] === "string" && text[1].toLowerCase() === info.charName.toLowerCase()) {
-							control.click();
-							this.click(6, 627, 572, 128, 35);
+						if (text instanceof Array && typeof text[1] === "string") {
+							count++;
 
-							break;
+							if (text[1].toLowerCase() === info.charName.toLowerCase()) {
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js]: should we be using .toLowerCase() in the other name checks?

File: d2bs/kolbot/libs/OOG.js (hunk 1 above)
Comment (imbalanced) [reply d2bs/kolbot/libs/OOG.js]: fixed as requested

File: d2bs/kolbot/libs/OOG.js (hunk 2 above)
Comment (imbalanced) [reply d2bs/kolbot/libs/OOG.js]: not necessary here. fixed

File: d2bs/kolbot/libs/OOG.js (hunk 3 above)
Comment (imbalanced) [reply d2bs/kolbot/libs/OOG.js]: added lowercase check for findCharacter function

File: d2bs/kolbot/libs/OOG.js [hunk 4]
```diff
@@ -975,15 +1028,31 @@ MainLoop:
...
-							break;
+								break MainLoop;
+							}
 						}
 					} while (control.getNext());
 				}
 
+				if (count === 8 || count === 16) { // check for additional characters up to 24
+					control = getControl(4, 237, 457, 72, 93);
+
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js]: Do we need mouseblock here?

Comment (imbalanced) [reply d2bs/kolbot/libs/OOG.js]: Yes, we don't want people moving mouse around to click somewhere else

Comment (noah-) [reply d2bs/kolbot/libs/OOG.js]: This section is missing a mouse block

Comment (imbalanced) [reply d2bs/kolbot/libs/OOG.js]: The function already has mouse block set in the beginning.

**Comment by bfass** [comment]
Would be nice to have this work on single player too

**Pull request merged**